pkg/trisa/mtls/mtls_test.go

```diff
@@ -0,0 +1,77 @@
+package mtls_test
+
+import (
+	"crypto/tls"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"github.com/trisacrypto/trisa/pkg/trisa/mtls"
+	"github.com/trisacrypto/trisa/pkg/trust"
+	"github.com/trisacrypto/trisa/pkg/trust/mock"
+	"google.golang.org/grpc"
+	"software.sslmate.com/src/go-pkcs12"
+)
+
+// Test that Config returns a valid TLS config.
+func TestConfig(t *testing.T) {
+	// Create private provider using the mock certificate chain
+	pfxData, err := mock.Chain()
+	require.NoError(t, err)
+	private, err := trust.Decrypt(pfxData, pkcs12.DefaultPassword)
+	require.NoError(t, err)
+
+	// Public provider should return an error
+	public := private.Public()
+	_, err = mtls.Config(public, trust.NewPool())
+	require.Error(t, err)
+
+	// Valid config
+	cfg, err := mtls.Config(private, trust.NewPool())
+	require.NoError(t, err)
+	require.NotNil(t, cfg)
+	require.Len(t, cfg.Certificates, 1)
+	require.Equal(t, uint16(tls.VersionTLS12), cfg.MinVersion)
+	require.NotEmpty(t, cfg.CurvePreferences)
+	require.True(t, cfg.PreferServerCipherSuites)
+	require.NotEmpty(t, cfg.CipherSuites)
+	require.Equal(t, tls.RequireAndVerifyClientCert, cfg.ClientAuth)
+	require.NotNil(t, cfg.ClientCAs)
+}
+
+// Test that ServerCreds returns a grpc.ServerOption for mtls.
+func TestServerCreds(t *testing.T) {
+	// Create private provider using the mock certificate chain
+	pfxData, err := mock.Chain()
+	require.NoError(t, err)
+	private, err := trust.Decrypt(pfxData, pkcs12.DefaultPassword)
+	require.NoError(t, err)
+
+	// Public provider should return an error
+	public := private.Public()
+	_, err = mtls.ServerCreds(public, trust.NewPool())
+	require.Error(t, err)
+
+	// Succesfully retuning a grpc.ServerOption
+	opt, err := mtls.ServerCreds(private, trust.NewPool())
```

Comment:
Interesting - does this mean that `ServerCreds` is essentially a wrapper for `Config`? I'm new to mTLS and trying to get a sense of the design patterns.

Reply:
I think that's a valid way of thinking about it. The way I interpreted the purpose of the code is that we're enabling mTLS via gRPC configuration, and `ServerCreds` and `ClientCreds` are helpers for returning gRPC options that we can use when creating a gRPC server or client. There's actually an example of this in the new GDS [members](https://github.com/trisacrypto/directory/blob/6a494cff0f4a58edcff88d6df54b240d2dcec3e5/pkg/gds/members.go#L56) code that was added a few days ago.